### ISSUE TITLE
Change in behavior of 'indent' rule in 4.18.0

### ISSUE THREAD

**Issue by zbjornson** [eslint/eslint#9995]
**Tell us about your environment**

* **ESLint Version:** 4.18.0
* **Node Version:** 8.9.4
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default with `ecmaVersion: 8`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
extends:
  - eslint:recommended
parserOptions:
  ecmaVersion: 8
rules:
  indent:
    - error
    - tab
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

This was valid in 4.17.0:

```js
function baz2() {
	return true &&
		![].some(g => {
			return true;
		});
}
```

but 4.18.0 wants it formatted like:

```js
function baz2() {
	return true &&
		![].some(g => {
		return true;
	});
}
```

This only happens if there's a `!` in front of the `[].some` bit.

```bash
eslint file.js
```

**What did you expect to happen?**
Same behavior as 4.17.0, per above.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  4:1   error  Expected indentation of 2 tabs but found 3  indent
  5:1   error  Expected indentation of 1 tab but found 2   indent
```

**Comment by not-an-aardvark** [comment]
Thanks for the report, I can reproduce this issue.
